astropy/stats/sigma_clipping.py

```diff
@@ -811,7 +904,7 @@ def sigma_clipped_stats(data, mask=None, mask_value=None, sigma=3.0,
                         sigma_upper=sigma_upper, maxiters=maxiters,
                         cenfunc=cenfunc, stdfunc=stdfunc, grow=grow)
     data_clipped = sigclip(data, axis=axis, masked=False, return_bounds=False,
-                           copy=False)
+                           copy=True)
```

Comment:
This might have been an oversight before because with ``False`` it modifies the input data - not entirely sure why this didn't come up before.

Reply:
Hmm, I'm not seeing that the input data are modified.  Simple example:

```
from astropy.stats import sigma_clip
rng = np.random.default_rng(seed=123)
data = rng.normal(10, 5, size=(500, 500))
data2 = data.copy()
data_clipped = sigma_clip(data, masked=False, return_bounds=False, axis=1, copy=False)
```
data and data2 are still the same (i.e. input doesn't appear to be modified).  data_clipped and data are different.

Reply:
The `copy` keyword has no effect unless `masked=True`.  From the docs:

<pre>
copy : bool, optional
        If `True`, then the ``data`` array will be copied.  If `False`
        and ``masked=True``, then the returned masked array data will
        contain the same array as the input ``data`` (if ``data`` is a
        `~numpy.ndarray` or `~numpy.ma.MaskedArray`).  The default is
        `True`.

Reply:
The docstring doesn't suggest to me that it has no effect if masked is False, to me it means more that:

* If `masked=False` the data will not be copied (this is the 'obvious' behavior a user would expect with this option)
* If `masked=True` the resulting array will be a different object since it will be a masked array but the data part of that will be the same (this is a little less intuitive hence why it is described specifically)

So I think this was actually a bug in the previous implementation - do you agree?

Reply:
(to be clear, the example you showed indeed copies anyway in main but not with this PR)